pkg/controller/provider/web/vsphere/vm.go

```diff
@@ -69,6 +69,20 @@ func (h VMHandler) List(ctx *gin.Context) {
 	}
 	pb := PathBuilder{DB: db}
 	for _, m := range list {
+		if err = db.Get(&m); err != nil {
```

Comment:
Nope, this will result of creating new DB query per each VM during the listing of all VMs.
Please checkout the `ListOptions` in the `db.List`